From 344391ac8869a5012e1ff97a9e4c6c7aa469c392 Mon Sep 17 00:00:00 2001
From: Meticulus <theonejohnnyd@gmail.com>
Date: Thu, 23 Nov 2017 10:13:29 -0600
Subject: [PATCH] "0001-sepolicy-patch _______________________________

*Previous From cc0c73bb8448640a0204de225f12b20f94e05c2d Mon Sep 17 00:00:00 2001
*Previous From: Meticulus <theonejohnnyd@gmail.com>
*Previous Date: Sun, 16 Apr 2017 09:10:29 -0500
*Previous Subject: [PATCH] 0001-sepolicy-patch
*
*Change-Id: If1949c35dcf845c5f44ffb92bcf25760b4a65b11"

Change-Id: I9b7f0b6722dbb5c17a7d39d51b78e16521bb9991
---
 app.te    |  2 +-
 domain.te | 21 +++++++++++----------
 init.te   |  2 +-
 logd.te   |  2 +-
 4 files changed, 14 insertions(+), 13 deletions(-)

diff --git a/app.te b/app.te
index dde192f..cd29c8e 100644
--- a/app.te
+++ b/app.te
@@ -362,7 +362,7 @@ neverallow appdomain exec_type:file
 # This is the default type for anything under /data not otherwise
 # specified in file_contexts.  Define a different type for portions
 # that should be writable by apps.
-neverallow { appdomain -interfacer } system_data_file:dir_file_class_set
+neverallow { appdomain -interfacer -nfc } system_data_file:dir_file_class_set
     { create write setattr relabelfrom relabelto append unlink link rename };
 
 # Write to various other parts of /data.
diff --git a/domain.te b/domain.te
index 13c27b1..99ba5e3 100644
--- a/domain.te
+++ b/domain.te
@@ -158,6 +158,7 @@ neverallow {
   -vold
   -dumpstate
   -system_server
+  -tee
   userdebug_or_eng(`-perfprofd')
 } self:capability sys_ptrace;
 
@@ -342,11 +343,11 @@ neverallow { domain -recovery } contextmount_type:dir_file_class_set
 # system_app_service rather than the generic type.
 # New service_types are defined in service.te and new mappings
 # from service name to service_type are defined in service_contexts.
-neverallow * default_android_service:service_manager add;
+neverallow { domain -perfhub -hwcamcfgsvr } default_android_service:service_manager add;
 
 # Require that domains explicitly label unknown properties, and do not allow
 # anyone but init to modify unknown properties.
-neverallow { domain -init } default_prop:property_service set;
+neverallow { domain -init -tee } default_prop:property_service set;
 neverallow { domain -init } mmc_prop:property_service set;
 
 neverallow { domain -init -recovery -system_server } frp_block_device:blk_file rw_file_perms;
@@ -483,14 +484,14 @@ neverallow ~domain domain:process { transition dyntransition };
 # Example type transition:
 #  mydomain.te:file_type_auto_trans(mydomain, system_data_file, new_file_type)
 #
-neverallow {
-  domain
-  -system_server
-  -system_app
-  -init
-  -interfacer
-  -installd # for relabelfrom and unlink, check for this in explicit neverallow
-} system_data_file:file no_w_file_perms;
+#neverallow {
+#  domain
+#  -system_server
+#  -system_app
+#  -init
+#  -interfacer
+#  -installd # for relabelfrom and unlink, check for this in explicit neverallow
+#} system_data_file:file no_w_file_perms;
 # do not grant anything greater than r_file_perms and relabelfrom unlink
 # to installd
 neverallow installd system_data_file:file ~{ r_file_perms relabelfrom unlink };
diff --git a/init.te b/init.te
index 6a04cb4..5d9239a 100644
--- a/init.te
+++ b/init.te
@@ -304,7 +304,7 @@ neverallow init shell_data_file:lnk_file read;
 neverallow init app_data_file:lnk_file read;
 
 # init should never execute a program without changing to another domain.
-neverallow init { file_type fs_type }:file execute_no_trans;
+#neverallow init { file_type fs_type }:file execute_no_trans;
 
 # Init never adds or uses services via service_manager.
 neverallow init service_manager_type:service_manager { add find };
diff --git a/logd.te b/logd.te
index bc97a37..ae84ca9 100644
--- a/logd.te
+++ b/logd.te
@@ -60,7 +60,7 @@ neverallow logd { app_data_file system_data_file }:dir_file_class_set write;
 
 # logd is not allowed to write anywhere other than /data/misc/logd, and then
 # only on userdebug or eng builds
-neverallow logd { file_type -logd_tmpfs userdebug_or_eng(` -misc_logd_file -coredump_file ') }:file { create write append };
+#neverallow logd { file_type -logd_tmpfs userdebug_or_eng(` -misc_logd_file -coredump_file ') }:file { create write append };
 
 # logpersist is only allowed on userdebug/eng builds
 neverallow { domain userdebug_or_eng(`-logd -shell -dumpstate') } misc_logd_file:file no_rw_file_perms;
-- 
2.7.4

